tests/tests_fabric/helpers/models.py

```diff
@@ -55,10 +56,15 @@ def after_optimizer_step(self, model: Module, optimizer: Optimizer) -> None:
 
     def run(self) -> None:
         model = self.get_model()
-        optimizer = self.get_optimizer(model)
-        dataloader = self.get_dataloader()
+        if isinstance(self.strategy, FSDPStrategy):
+            model = self.setup_module(model)
+            optimizer = self.get_optimizer(model)
+            optimizer = self.setup_optimizers(optimizer)
```

Comment:
I liked the boring implementation, and the fact that the test had the special order of setup spelled out explicitly. 

but ok, perhaps in pt 2.0 we will be able to remove the if-else.